### PR TITLE
graphql: add DeleteWebhook GraphQL resolver

### PR DESCRIPTION
**Pull request by sashaostrikov**
Test plan:
GraphQL schema check and unit tests added

Closes https://github.com/sourcegraph/sourcegraph/issues/42862

### CODE REVIEW

**Review by mrnugget — APPROVED** [review]

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.00% (0.00 kb) | 0.00% (0.00 kb) | 0.00% (0.00 kb) | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/ao/webhooks/gql-delete-mutation/compare-report.html;tab=live_object) for a full comparison between the commits d872cb2 and afbccc5 or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

File: cmd/frontend/graphqlbackend/schema.graphql [hunk 1]
```diff
@@ -108,6 +108,11 @@ type Mutation {
     """
     createWebhook(codeHostKind: String!, codeHostURN: String!, secret: String): Webhook!
 
+    """
+    Deletes a webhook by given UUID. Only site admins may perform this mutation.
+    """
+    deleteWebhook(uuid: String!): EmptyResponse!
```
[the remark this answers is not in the document]
Comment (sashaostrikov) [reply cmd/frontend/graphqlbackend/schema.graphql:114]: Yeah, UUID usage is a prediction/speculation for future auto-sync times where UUID will be needed if the syncs will be based on URLs.

We have both "get by" methods for ID and UUID at the moment, should we have 2 delete methods as well?

[the remark this answers is not in the document]
Comment (mrnugget) [reply cmd/frontend/graphqlbackend/schema.graphql:114]: > Yeah, UUID usage is a prediction/speculation for future auto-sync times where UUID will be needed if the syncs will be based on URLs.

But those won't go through the GraphQL API, no?

> We have both "get by" methods for ID and UUID at the moment, should we have 2 delete methods as well?

You can have a single method with an `DeleteWebhookOpts` parameter and then do a `if opts.ID != 0` check etc.

[the remark this answers is not in the document]
Comment (sashaostrikov) [reply cmd/frontend/graphqlbackend/schema.graphql:114]: > But those won't go through the GraphQL API, no?

That's a good point, yep.

I'm gonna change the db layer for that first, then reopen this PR, I don't want to include all the changes in what is supposed to be a gql layer commit

**Comment by sashaostrikov** [comment]
This PR will be back as soon as https://github.com/sourcegraph/sourcegraph/issues/43080 is implemented

**Pull request closed without merging**